app/src/modules/settings/routes/roles/permissions-detail/components/presets.vue

```diff
@@ -47,7 +55,31 @@ export default defineComponent({
 			},
 		});
 
-		return { t, presets };
+		const relationsStore = useRelationsStore();
+		const relations = relationsStore.getRelationsForCollection(props.permission.collection);
+		const relationFields = relations.map((relation) => relation.field);
```

Comment:
Does this correctly filter out just the `o2m` relationships, or does this affect all relational fields?

Reply:
Sorry, I did not do proper testing yesterday evening. I have changed it now and I think now it behaves correctly. I tested for m2m, o2m and m2o with different configurations. I also updated the comment to explain the motivation behind the warning more clearly. Maybe the warning text itself could be extended to be more precise, e.g.:
Relational [...] syntax. Otherwise, if a relational app interface is used, it will not show the preset correctly.

The warning is shown for all relations where the meta.one_field is set now. For the o2m relation, the warning is also shown as the interface does not support the array syntax either. For m2o, there is no warning as the relation is filtered out. But even if not, the warning would only be shown when a non-empty array is entered which does not make sense for a m2o field. Validation for the whole preset could be added, but I think there are more urgent topics at the moment.